backend/src/ml_pipeline/helper_functions.py

```diff
@@ -158,6 +159,40 @@ def timeseries_pred(file, meal_data, plotname):
     plt.tight_layout()
     plt.savefig(plotname)
 
+def timeseries_pred_plotly(file, meal_data, plotname):
+    # Create figure with secondary y-axis
+    fig = make_subplots(specs=[[{"secondary_y": True}]])
+
+    fig.add_trace(go.Line(x=file['timestamps'], y=file['CGM'],  # title='Raw CGM Data with Identified Meals',
+                          # markers=True,
+                          name='Glucose (mg/dL)'
+                          ))
+
+    # fig.update_shapes(dict(xref='Glucose (mg/dL)'))
+    fig.update_layout(
+        xaxis_title="Time", yaxis_title="Glucose (mg/dL)",
+        title={
+            'text': 'Raw CGM Data with Identified Meals',
+            'x': 0.5,
+            'xanchor': 'center',
+            'yanchor': 'top'
+        })
+
+    fig.update_traces(line={'width': 5})
+    fig.add_trace(go.Scatter(x=file.timestamps, y=np.repeat(180, len(file.timestamps)), name='Target Glucose Range',
+                             line=dict(color='green', width=4, dash='dash')))
+    fig.add_trace(go.Scatter(x=file.timestamps, y=np.repeat(70, len(file.timestamps)), showlegend=False,
+                             line=dict(color='green', width=4, dash='dash')))
+
+    fig.add_trace(go.Bar(x=meal_data[meal_data.CHO_total > 0]['start_block'],
+                         y=meal_data[meal_data.CHO_total > 0]["carb_preds"],
+                         marker={'color': 'red'},
+                         name='Meal - Carbs (g)',
+
+                         ),
+                  # secondary_y=True,
+                  )
+    fig.write_image(plotname)
 
```

Comment:
This is okay in some instances, but we want it to be identical to how its generated in the `generate_plot` function like so:

```# Convert the figure to an HTML string
    fig_html = py.to_html(fig, full_html=True)

    # Save the HTML string as an HTML file
    with open(filename, 'w') as f:
        f.write(fig_html)
```

Going to add this in the code but commenting for reference